### ISSUE TITLE
Allow Uncontrolled Components

### ISSUE THREAD

**Issue by vixalien** [Shopify/polaris-react-archive#11232]
## Feature request summary

Components like `TextField` and `Select` only work when they are controlled, i.e. they need the properties `value` and `onChange` be set so that their state can be managed by javascript code.

## Rationale

By allowing form inputs to be uncontrolled, we could eliminate those props and ensure Polaris Forms works without JavaScript. This is particularly important in Remix, where the browser should be delegated to do everything it can and use JS when necessary.

---

🌟 Feature requests that are not yet planned will be closed. We then use the issue’s :+1: upvotes to track and set priorities. See the [contribution guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md) for more information.

**Comment by davidhousedev** [comment]
Another dimension to this is compatibility with the App Bridge components. For example, the Contextual Save Bar relies on native HTML for events and states in order to work properly, but it's challenging to emulate the required events when you're using controlled inputs.

Being able to use uncontrolled inputs would ensure that all native DOM events get sent by-default.

**Comment by mkromann** [comment]
This would be lovely — especially considering that Remix is such a breeze with uncontrolled components.

**Comment by vixalien** [comment]
This is still relevant…

**Comment by grundmanise** [comment]
Agree. I don't see a reason why they should be controlled components only.

**Comment by JannikWempe** [comment]
This should have a high prio after Remix has become the standard. 

Remix works very well with uncontrolled components and common libraries like `conform` rely on them.

Adding support for uncontrolled inputs would make development a lot easier.

**Comment by ryanolf** [comment]
+1

**Comment by husseyexplores** [comment]
Getting more relevant day by day with Remix.
Consider implementing this please .

**Comment by dizzyjaguar** [comment]
Any update on this? This seems like a given being that **Remix** is the suggested way to create Shopify Apps
